src/platform/qt/library/LibraryModel.cpp

```diff
@@ -0,0 +1,422 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ * Copyright (c) 2013-2022 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "LibraryModel.h"
+
+#include "utils.h"
+
+#include <QSortFilterProxyModel>
+#include <QItemSelectionModel>
+#include <QApplication>
+#include <QDir>
+#include <algorithm>
+
+using namespace QGBA;
+
+namespace QGBA {
+
+struct SpanSet {
```

Comment:
I wonder if some version of this (perhaps with templates) might be useful in qt/utils.h. We can think about that later, though.